src/tds/data.c

```diff
@@ -1476,6 +1476,95 @@ tds_sybbigtime_put(TDSSOCKET *tds, TDSCOLUMN *col, int bcp7)
 	return TDS_SUCCESS;
 }
 
+TDSRET
+tds_mstabletype_get_info(TDSSOCKET * tds, TDSCOLUMN * col)
+{
+	/* Table type is strictly only an input variable */
+	return TDS_FAIL;
+}
+
+TDS_INT
+tds_mstabletype_row_len(TDSCOLUMN *col)
+{
+	return sizeof(TDS_TVP);
+}
+
+TDSRET
+tds_mstabletype_get(TDSSOCKET * tds, TDSCOLUMN * col)
+{
+	/* Table type is strictly only an input variable */
+	return TDS_FAIL;
+}
+
+TDSRET
+tds_mstabletype_put_info(TDSSOCKET * tds, TDSCOLUMN * col)
+{
+	TDS_TVP * table = (TDS_TVP *) col->column_data;
+
+	/* TVP_TYPENAME */
+	tds_put_byte(tds, 0); /* Empty DB name */
+	tds_put_byte(tds, strlen(table->schema));
+	tds_put_string(tds, table->schema, -1);
```

Comment:
When calling `SQLBindParameter`, we will also pass in the C source type (`SQL_C_CHAR` or `SQL_C_WCHAR` for strings) along with the pointer to the buffer, which allows us to determine how to handle the buffer's contents when performing coversions. However, we don't pass in such a value when binding table values. Would it be ok for us to require that the TVP type name will always passed as a pointer to a `char`/`wchar` buffer?

Reply:
No, everything that gets into FreeTDS should be converted to UTF-8 (well, for ODBC and with some exceptions). So you should not having to do it. There are many examples in `src/tds/query.c`, like
```c
    TDS_START_LEN_TINYINT(tds) {
        tds_put_string(tds, dyn->id, -1);
    } TDS_END_LEN
```
(yes, it may a bit weird.. but works)

Reply:
Thanks for clarifying. However, `tds_freeze_written` function returns the total number of bytes written to the socket, and in the case of the table type name, this could be more than the actual number of characters in the string if we use a multibyte encoding. There are instances in the codebase where we make some adjustments to the number of bytes before passing it to `tds_freeze_close_len` (e.g. sending the procname of an RPC below).

https://github.com/FreeTDS/freetds/blob/3a16761a8e39d3b289fee0356ae7116be686713e/src/tds/query.c#L2007

Would it be safe to do the same thing when sending the TVP type name? E.g.
```c++
	TDSFREEZE current_freeze[1];
	size_t written;

	tds_freeze(tds, current_freeze, 1);
	tds_put_string(tds, table->schema, -1);
	written = tds_freeze_written(current_freeze) / 2;
	tds_freeze_close_len(current_freeze, written);
```